libsolidity/analysis/ControlFlowGraph.cpp

```diff
@@ -34,15 +34,35 @@ bool CFG::constructFlow(ASTNode const& _astRoot)
 
 bool CFG::visit(FunctionDefinition const& _function)
 {
-	if (_function.isImplemented())
-		m_functionControlFlow[&_function] = ControlFlowBuilder::createFunctionFlow(m_nodeContainer, _function);
+	if (_function.isImplemented() && _function.isFree())
+		m_functionControlFlow[pair(nullptr, &_function)] = ControlFlowBuilder::createFunctionFlow(m_nodeContainer, _function, nullptr);
 	return false;
 }
 
-FunctionFlow const& CFG::functionFlow(FunctionDefinition const& _function) const
+bool CFG::visit(ContractDefinition const& _contract)
 {
-	solAssert(m_functionControlFlow.count(&_function), "");
-	return *m_functionControlFlow.find(&_function)->second;
+	set<CallableDeclaration const*> overridden;
+
+	for (ContractDefinition const* contract: _contract.annotation().linearizedBaseContracts)
+		for (FunctionDefinition const* function: contract->definedFunctions())
+		{
+			if (overridden.count(function))
```

Comment:
Are you sure we can ignore those? Maybe we need the function call graph here, too. Test case:
```
contract A {
  function f() public virtual returns (uint) { g(); }
  function g() internal virtual { revert(); }
}
contract B is A {
  function f() pubic override { A.f() }
  function g() internal override {}
}
```

Reply:
Did you add this test case?

Reply:
I forgot. Will add now. Does not cause any error, but I think it should because `A.f()` would call `B.g()`, correct?

Reply:
Yes